File: sdk/nodejs/src/resources/billing.js
"use strict";

const { BaseResource } = require("./base");

class BillingResource extends BaseResource {
  async getBalance() {
    return this._http.get("/api/balance");
  }

  async listInvoices(params = {}) {
    return this._http.get("/api/invoice");
  }

  async getInvoice(invoiceId) {
    return this._http.get("/api/invoice/{id}");
  }

  async applyCredit(invoiceId, data) {
    return this._http.post("/api/invoice/{id}/credit", data);
  }

  async listPayments(params = {}) {
    return this._http.get("/api/payment");
  }

  async getPaymentFees() {
    return this._http.get("/api/payment/fees");
  }

}

module.exports = { BillingResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Billing.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Billing extends BaseResource {

    public Billing(HttpClient http) {
        super(http);
    }

    /** Get balance */
    public Map<String, Object> balance() throws Exception {
        return http.get("/api/balance");
    }

    /** List invoices */
    public Map<String, Object> invoices() throws Exception {
        return http.get("/api/invoice");
    }

    /** Get invoice */
    public Map<String, Object> invoice(int id) throws Exception {
        return http.get(String.format("/api/invoice/%id", id));
    }

    /** Credit invoice */
    public Map<String, Object> invoiceCredit(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/invoice/%id/credit", id), data);
    }

    /** List payments */
    public Map<String, Object> payments() throws Exception {
        return http.get("/api/payment");
    }

    /** Get payment fees */
    public Map<String, Object> paymentFees() throws Exception {
        return http.get("/api/payment/fees");
    }

}
